mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/model/impl/JavaBeanTypeModel.java

```diff
@@ -99,46 +96,41 @@ public boolean isSubTypeOf(Class<?> superClassCandidate) {
 	}
 
 	@Override
-	@SuppressWarnings( "unchecked" )
-	public Stream<? extends PojoRawTypeModel<? super T>> getAscendingSuperTypes() {
-		return JavaClassOrdering.get().getAscendingSuperTypes( clazz )
-				.map( clazz -> introspector.getTypeModel( (Class<? super T>) clazz ) );
+	public Stream<JavaBeanTypeModel<? super T>> getAscendingSuperTypes() {
+		return introspector.getAscendingSuperTypes( xClass );
 	}
 
 	@Override
-	@SuppressWarnings( "unchecked" )
 	public Stream<? extends PojoRawTypeModel<? super T>> getDescendingSuperTypes() {
-		return JavaClassOrdering.get().getDescendingSuperTypes( clazz )
-				.map( clazz -> introspector.getTypeModel( (Class<? super T>) clazz ) );
+		return introspector.getDescendingSuperTypes( xClass );
 	}
 
 	@Override
 	public <A extends Annotation> Optional<A> getAnnotationByType(Class<A> annotationType) {
-		return introspector.getAnnotationByType( clazz, annotationType );
+		return introspector.getAnnotationByType( xClass, annotationType );
 	}
 
 	@Override
 	public <A extends Annotation> Stream<A> getAnnotationsByType(Class<A> annotationType) {
-		return introspector.getAnnotationsByType( clazz, annotationType );
+		return introspector.getAnnotationsByType( xClass, annotationType );
 	}
 
 	@Override
 	public Stream<? extends Annotation> getAnnotationsByMetaAnnotationType(Class<? extends Annotation> metaAnnotationType) {
-		return introspector.getAnnotationsByMetaAnnotationType( clazz, metaAnnotationType );
+		return introspector.getAnnotationsByMetaAnnotationType( xClass, metaAnnotationType );
 	}
 
 	@Override
 	public PojoPropertyModel<?> getProperty(String propertyName) {
 		String normalizedName = Introspector.decapitalize( propertyName );
```

Comment:
Actually it looks like you didn't remove *all* uses of `java.beans.Introspector`...

I think this line can be removed and all tests will pass.

There's also a reference to the `java.beans` package in `org.hibernate.search.integrationtest.mapper.pojo.mapping.definition.DependencyIT`, mainly because we have an `@Transient` annotation somewhere in `derivedFrom_error_missingPath`. I think this annotation can be removed safely, too.

However, we really should do something to check at build time that we don't use `java.beans`, as it will be removed from future versions of the JDK.
Maybe you could add a rule to `src/main/resources/forbidden-runtime.txt` and `src/main/resources/forbidden-tests.txt`?

Reply:
Sure